### PR TITLE
add setSubsets in fractalis service

### PR DESCRIPTION
**Pull request by beaugogh**

### CODE REVIEW

File: src/app/services/fractalis.service.ts [hunk 1]
```diff
@@ -157,9 +175,30 @@ export class FractalisService {
...
+  private configSubsetsUpdate() {
+    this.cohortService.cohortsUpdated.asObservable()
+      .subscribe((cohorts: Cohort[]) => {
+        let subjectCalls = [];
+        cohorts
+          .filter((cohort: Cohort) => {
+            return cohort.selected;
+          })
+          .forEach((cohort: Cohort) => {
+            subjectCalls.push(this.resourceService.getSubjects(cohort.constraint));
```
Comment (gijskant) [inline src/app/services/fractalis.service.ts]: Does this fetch patient ids for all cohorts whenever one of the cohorts has changed or selected?

File: src/app/services/fractalis.service.ts [hunk 2]
```diff
@@ -157,9 +175,30 @@ export class FractalisService {
...
+          });
+        let idSets = [];
+        forkJoin(subjectCalls)
+          .subscribe(res => {
+            for (let i = 0; i < subjectCalls.length; i++) {
+              const ids = res[i].map((subject: TransmartPatient) => {
+                return subject.id;
+              });
+              idSets.push(ids);
```
Comment (ewelinagr) [inline src/app/services/fractalis.service.ts]: Looks like this way there may be some duplications of ids on the `idSets` list

File: src/app/services/fractalis.service.ts (hunk 1 above)
Comment (beaugogh) [reply src/app/services/fractalis.service.ts]: yes, whenever one of the cohorts is changed, the cohortsUpdated subject will emit event to inform its subscribers, in this case, the only subscriber here in fractalis service, catches it and does subsets update

File: src/app/modules/gb-side-panel-module/accordion-components/gb-cohorts/gb-cohorts.component.ts [hunk 3]
```diff
@@ -217,7 +217,8 @@ export class GbCohortsComponent implements OnInit {
 
   onCohortCheckClick(e: MouseEvent) {
     e.stopPropagation();
-    this.cohortService.updateAll();
+    this.cohortService.cohortsUpdated.next(this.cohortService.cohorts);
+    this.cohortService.updateCountsWithAllCohorts();
```
[the remark this answers is not in the document]
Comment (beaugogh) [reply src/app/modules/gb-side-panel-module/accordion-components/gb-cohorts/gb-cohorts.component.ts]: the cohortsUpdated subject will do the update inside updateCountsWithAllCohorts

[the remark this answers is not in the document]
Comment (beaugogh) [reply src/app/modules/gb-side-panel-module/accordion-components/gb-cohorts/gb-cohorts.component.ts]: somehow the change is not reflected here, see https://github.com/thehyve/glowing-bear/blob/mcohorts/src/app/modules/gb-side-panel-module/accordion-components/gb-cohorts/gb-cohorts.component.ts

**Pull request merged**